packages/web/components/trade-clipboard/index.tsx

```diff
@@ -0,0 +1,481 @@
+                        ? "text-white-full"
+                        : "text-white-disabled"
+                    )}
+                  >{`≈ ${tradeTokenInConfig.expectedSwapResult.amount
+                    .trim(true)
+                    .shrink(true)
+                    .maxDecimals(6)
+                    .toString()}`}</h5>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div className="mt-[1.125rem] border border-white-faint rounded-lg bg-card py-3 px-4">
+            <div className="flex justify-between">
+              <div className="subtitle2 text-wireframes-lightGrey">Rate</div>
+              <div className="flex flex-col gap-y-1.5 text-right">
+                <div className="subtitle2 text-wireframes-lightGrey">
+                  {`1 ${
+                    tradeTokenInConfig.sendCurrency.coinDenom
+                  } = ${tradeTokenInConfig.expectedSwapResult.beforeSpotPriceWithoutSwapFeeOutOverIn
+                    .trim(true)
+                    .maxDecimals(3)
+                    .toString()} ${tradeTokenInConfig.outCurrency.coinDenom}`}
+                </div>
+                <div className="caption text-wireframes-grey">
+                  {`1 ${
+                    tradeTokenInConfig.outCurrency.coinDenom
+                  } = ${tradeTokenInConfig.expectedSwapResult.beforeSpotPriceWithoutSwapFeeInOverOut
+                    .trim(true)
+                    .maxDecimals(3)
+                    .toString()} ${tradeTokenInConfig.sendCurrency.coinDenom}`}
+                </div>
+              </div>
+            </div>
+            <div className="flex justify-between mt-2.5">
+              <div className="subtitle2 text-wireframes-lightGrey">
+                Swap Fee
+              </div>
+              <div className="subtitle2 text-wireframes-lightGrey">
+                {tradeTokenInConfig.expectedSwapResult.swapFee.toString()}
+              </div>
+            </div>
+            <div className="flex justify-between pt-4 mt-4 border-t border-white-faint">
+              <div className="subtitle2 text-white-high">
+                Estimated Slippage
+              </div>
+              <div className="subtitle2 text-white-high">
+                {tradeTokenInConfig.expectedSwapResult.slippage.toString()}
+              </div>
+            </div>
+          </div>
+
+          {/* TODO: Styling warning/error case. Render loading state. Handle rejection. */}
+          <button
+            className="mt-[1.125rem] flex justify-center items-center w-full h-[3.75rem] rounded-lg bg-primary-200 text-white-full text-base font-medium shadow-md"
+            disabled={
+              tradeTokenInConfig.error != null &&
+              tradeTokenInConfig.optimizedRoutePaths.length > 0
+            }
+            onClick={async (e) => {
+              e.preventDefault();
+
+              if (tradeTokenInConfig.optimizedRoutePaths.length > 0) {
```

Comment:
Seems like common logic, should this live on a method on `tradeTokenInConfig`'s class?